Foi criado um arquivo chamado limpeza_base_varejo.ipynb onde será realizado a limpeza e tratamento dos dados

Depois de realizar a criação do ambiente vistual e installar as bibliotecas pandas, numpy, matplotlib e seaborn, será importado o arquivo de dados "dados_varejo.csv" para análise e tratamento.

Sera importados as bibliotecas pandas para manipulação de dados, numpy para operações matemáticas, matplotlib e seaborn para visualização de dados. Em seguida, será carregado o arquivo CSV em um DataFrame do pandas para iniciar o processo de limpeza e tratamento dos dados.

Descrição das tabelas:

1. DATA: Data da compra;
2. CO_ID: Identificação do número de compra (número da nota fiscal);
3. CL_ID: Identificação do cliente (número do cliente);
4. CL_GENERO: Sexo biológico informado pelo cliente;
2
5. CL_EC: Estado civil do cliente:
1: Casado ou união estával;
2: Divorciado;
3: Separado;
4. Solteiro;
5: Viúvo.
6. CL_FHL: Número de filhos do cliente;
7. CL_SEG: Segmentação econômica do cliente (classe A, B ou C);
8. PR_ID: Código do produto (SKU) adquirido;
9. PR_CAT: Categoria do produto adquirido;
10. PR_NOME: Nome do produto adquirido.

Importando as bibliotecas necessárias para a análise e tratamento dos dados:

In [151]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
import re
import datetime

In [152]:
#Carregando o arquivo CSV em um DataFrame do pandas para iniciar o processo de limpeza e tratamento dos dados.
df = pd.read_csv("Base Varejo.csv")
df.head()

,DATA;CO_ID;CL_ID;CL_GENERO;CL_EC;CL_FHL;CL_SEG;PR_ID;PR_CAT;PR_NOME;;;;
0,01/02/2019;1000;534;M;4;1;C;67;BEBIDAS;REFRIGE...
1,01/02/2019;1000;534;M;4;1;C;70;BEBIDAS;REFRIGE...
2,01/02/2019;1000;534;M;4;1;C;178;HIGIENE;LENCO ...
3,01/02/2019;1000;534;M;4;1;C;4;ALIMENTOS;ABACAX...
4,01/02/2019;1000;534;M;4;1;C;175;LIMPEZA;LIMPAD...


In [153]:
df.shape


(830000, 1)

In [154]:
df.columns

Index(['DATA;CO_ID;CL_ID;CL_GENERO;CL_EC;CL_FHL;CL_SEG;PR_ID;PR_CAT;PR_NOME;;;;'], dtype='str')

Depois de ter observado que os dados estavan confusos decidi fazer observar os dados com os metodos ( .shape e .columns) para poder entender porque os dados pareciam tão desorganizados . Identificando que aprensentava todos os nomes das colunas num único campo.

 Crie outro dataframe a partir de apenas as colunas mais importantes para poder trabalhar e analisar, escolhi: 'CO_ID','CL_ID','DATA','CL_GENERO', 'CL_EC', 'CL_FHL', 'PR_CAT', 'PR_NOME'

Para esta analise exploratoria de dados deste arquivo se tomaram apenas 100 linhas de registro. Para fazer para fines academicos e de aprendizado, pois o arquivo original contem mais de 1.000 linhas.

In [155]:
df_enxuto= pd.read_csv('Base Varejo.csv', sep=";", encoding='latin-1',usecols=['CO_ID','CL_ID','DATA','CL_GENERO', 'CL_EC', 'CL_FHL', 'CL_SEG', 'PR_CAT', 'PR_NOME'], nrows=100) # crie o codigo completo padrão para leitura passando o arquivo original para um dataframe chamado df_enxuto com separado por ; encodinf latin-1
df_enxuto.head()

,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_CAT,PR_NOME
0,01/02/2019,1000,534,M,4,1,C,BEBIDAS,REFRIGERANTE GUARANA
1,01/02/2019,1000,534,M,4,1,C,BEBIDAS,REFRIGERANTE OUTROS
2,01/02/2019,1000,534,M,4,1,C,HIGIENE,LENCO UMEDECIDO
3,01/02/2019,1000,534,M,4,1,C,ALIMENTOS,ABACAXI
4,01/02/2019,1000,534,M,4,1,C,LIMPEZA,LIMPADOR MULTIUSO


In [156]:
df_enxuto.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   DATA       100 non-null    str  
 1   CO_ID      100 non-null    int64
 2   CL_ID      100 non-null    int64
 3   CL_GENERO  100 non-null    str  
 4   CL_EC      100 non-null    int64
 5   CL_FHL     100 non-null    int64
 6   CL_SEG     100 non-null    str  
 7   PR_CAT     100 non-null    str  
 8   PR_NOME    100 non-null    str  
dtypes: int64(4), str(5)
memory usage: 7.2 KB


Se observou data em formato string e seran tratadas para o fotmato datetime.

In [173]:
df_enxuto.describe()

,DATA,CO_ID,CL_ID,CL_FHL
count,100,100.000000,100.000000,100.0000
mean,2019-01-02 00:00:00,1031.740000,415.230000,0.8200
min,2019-01-02 00:00:00,1000.000000,279.000000,0.0000
25%,2019-01-02 00:00:00,1000.000000,290.000000,0.0000
50%,2019-01-02 00:00:00,1000.000000,534.000000,1.0000
75%,2019-01-02 00:00:00,1078.000000,534.000000,1.0000
max,2019-01-02 00:00:00,1078.000000,534.000000,2.0000
std,NaN,35.395206,124.293222,0.6724


annalizando com o metodo describe()  na coluna CL_FHL (número de filhos do cliente) se observa que o valor minimo é 0, o valor maximo é 2, a media é 1.0 e a mediana é 1.0, o que indica que a maioria dos clientes tem 1 filho.

In [158]:
df_enxuto['CL_SEG'].value_counts()

CL_SEG
C    52
B    48
Name: count, dtype: int64

Na couluna CL_SEG se resalizou um value_counts() para observar a maior quantidade de compras foram feitas por clientes da classe econômica C, seguida da classe econômica B.

In [159]:
df_enxuto['DATA']= pd.to_datetime(df_enxuto['DATA'])
df_enxuto.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   DATA       100 non-null    datetime64[us]
 1   CO_ID      100 non-null    int64         
 2   CL_ID      100 non-null    int64         
 3   CL_GENERO  100 non-null    str           
 4   CL_EC      100 non-null    int64         
 5   CL_FHL     100 non-null    int64         
 6   CL_SEG     100 non-null    str           
 7   PR_CAT     100 non-null    str           
 8   PR_NOME    100 non-null    str           
dtypes: datetime64[us](1), int64(4), str(4)
memory usage: 7.2 KB


Verificando a quantidade de valores nulos e duplicados

In [160]:
df_enxuto.isnull().sum() #Não se observaram valores nulos 

DATA         0
CO_ID        0
CL_ID        0
CL_GENERO    0
CL_EC        0
CL_FHL       0
CL_SEG       0
PR_CAT       0
PR_NOME      0
dtype: int64

In [161]:
df_enxuto.duplicated().sum() #Se observaram duplicatas 

np.int64(17)

In [162]:
df_enxuto.drop_duplicates()

,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_CAT,PR_NOME
0,2019-01-02,1000,534,M,4,1,C,BEBIDAS,REFRIGERANTE GUARANA
1,2019-01-02,1000,534,M,4,1,C,BEBIDAS,REFRIGERANTE OUTROS
2,2019-01-02,1000,534,M,4,1,C,HIGIENE,LENCO UMEDECIDO
3,2019-01-02,1000,534,M,4,1,C,ALIMENTOS,ABACAXI
4,2019-01-02,1000,534,M,4,1,C,LIMPEZA,LIMPADOR MULTIUSO
...,...,...,...,...,...,...,...,...,...
92,2019-01-02,1078,290,F,1,0,B,LIMPEZA,TIRA MANCHA
93,2019-01-02,1078,290,F,1,0,B,ALIMENTOS,LATA DE MILHO
94,2019-01-02,1078,290,F,1,0,B,LIMPEZA,RODO
95,2019-01-02,1078,290,F,1,0,B,ALIMENTOS,SALGADINHO


In [163]:
df_enxuto.duplicated().sum()

np.int64(17)

Se intentou dropar as duplicatas sem sucesso

verificando valores unicos

In [164]:
df_enxuto[['CO_ID','CL_ID','DATA','CL_GENERO', 'CL_EC', 'CL_FHL', 'PR_CAT', 'PR_NOME']].nunique()

CO_ID         3
CL_ID         3
DATA          1
CL_GENERO     2
CL_EC         2
CL_FHL        3
PR_CAT        7
PR_NOME      69
dtype: int64

aplicando o metodo nunique() para verificar a quantidade de valores unicos em cada coluna do dataframe. Se obseva segundo os dados das colunas que a coluna CO_ID e CL_ID apresentam 3 valores o que dignifica que existem 3 compras diferentes e 3 clientes diferentes, enquanto as outras colunas apresentam apenas 1 valor unico, o que indica que todos os registros possuem o mesmo valor para essas colunas.

Se realizou tratamento da coluna de CL_EC que referese a Estado civil do cliente para trocar os numeros para seus respetiva descrição

In [165]:
df_enxuto['CL_EC']=df_enxuto['CL_EC'].replace(1, 'Casado ou união estável')
df_enxuto['CL_EC']=df_enxuto['CL_EC'].replace(2, 'Divorciado')
df_enxuto['CL_EC']=df_enxuto['CL_EC'].replace(3, 'Separado')
df_enxuto['CL_EC']=df_enxuto['CL_EC'].replace(4, 'Solteiro')
df_enxuto['CL_EC']=df_enxuto['CL_EC'].replace(5, 'Viúvo')

In [166]:
df_enxuto['CL_EC'] #verificando o resultado

0                    Solteiro
1                    Solteiro
2                    Solteiro
3                    Solteiro
4                    Solteiro
               ...           
95    Casado ou união estável
96    Casado ou união estável
97    Casado ou união estável
98    Casado ou união estável
99    Casado ou união estável
Name: CL_EC, Length: 100, dtype: object

In [167]:
df_enxuto.head()

,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_CAT,PR_NOME
0,2019-01-02,1000,534,M,Solteiro,1,C,BEBIDAS,REFRIGERANTE GUARANA
1,2019-01-02,1000,534,M,Solteiro,1,C,BEBIDAS,REFRIGERANTE OUTROS
2,2019-01-02,1000,534,M,Solteiro,1,C,HIGIENE,LENCO UMEDECIDO
3,2019-01-02,1000,534,M,Solteiro,1,C,ALIMENTOS,ABACAXI
4,2019-01-02,1000,534,M,Solteiro,1,C,LIMPEZA,LIMPADOR MULTIUSO


Creada copia do dataframe para poder trabalhar com as colunas con dados referentes ao cliente para tratar duplicatas

In [170]:
df_clientes= df_enxuto[['CO_ID','CL_ID','CL_GENERO'	,'CL_EC','CL_FHL','CL_SEG' ]].copy()
df_clientes.head()

,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG
0,1000,534,M,Solteiro,1,C
1,1000,534,M,Solteiro,1,C
2,1000,534,M,Solteiro,1,C
3,1000,534,M,Solteiro,1,C
4,1000,534,M,Solteiro,1,C


In [171]:
df_clientes.drop_duplicates() #dropando valores duplicados 

,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG
0,1000,534,M,Solteiro,1,C
52,1040,279,M,Casado ou união estável,2,B
67,1078,290,F,Casado ou união estável,0,B


Tirando as duplicatas do dataframe se consiguiu visualizar os 3 clientes que fizeram compras diferentes, e se observou que os clientes 1 e 2 são do sexo feminino e o cliente 3 é do sexo masculino. Se observou que os clientes 1 e 3 são solteiros e o cliente 2 é casado. Se observou que o client 1 não tem filhos, o cliente 2 tem 2 filhos e o cliente 3 tem 1 filho. Se observou que os clientes 1 e 3 pertencem a classe econômica C e o cliente 2 pertence a classe econômica A.

In [172]:
df_enxuto['CL_ID'].value_counts()


CL_ID
534    52
290    33
279    15
Name: count, dtype: int64

Insights obtidos a partir da análise exploratória dos dados da coluna CL_ID (Identificação do cliente): Podese ver que o 
    cliente 534 Masculino, Solteiro, Classe Econômica C realizou o maior numero de compras , seguido pelo
    cliente 290 Feminino, Casada, sem filhos classe economica B e depois o
    cliente 279 Masculino, Casado, com 2 filhos Classe Econômica B realizou o menor numero de compras.